### PR TITLE
Fixed #239

### LINKED ISSUE #239: Introduce separate batch sizes for train, validation, and test
The distribution of wav lengths in the train, validation, and test sets may be skewed. For example, if small wav files are in the training set and larger ones in the validation set.

If this is the case, then the maximum training batch size is too large for the validation set.

To solve this separate batch sizes should be introduced for the train, validation, and test sets.

Far from being theoretic, this skew exists in the TED data set. The largest wav files form the train, validation, and test sets are as follows:
```
train
-rw-rw-r-- 1 alissy alissy 634K Oct 10 05:47 JonathanHarris_2007P-391.71-411.97.wav

validation
-rw-rw-r-- 1 alissy alissy 1.3M Oct 10 05:47 WadeDavis_2003-123.31-163.23.wav

test
-rw-rw-r-- 1 alissy alissy 1018K Oct 10 05:50 JaneMcGonigal_2010-196.922-229.47.wav
```
As one can see, the largest wav in the validation data set is about twice the size of the largest wav in the test data set. A similar statement is true for the test data set. This causes the batch size 48 to work for the training set and fail for the validation set.

### PR DESCRIPTION
**Pull request by kdavis-mozilla**

### CODE REVIEW

File: DeepSpeech.ipynb [hunk 1]
```diff
@@ -1315,7 +1319,12 @@
     "    graph = tf.Graph()\n",
     "    with graph.as_default():\n",
     "        # Obtain all the data\n",
-    "        data_sets = ds_importer_module.read_data_sets(ds_dataset_path, batch_size, n_input, n_context)\n",
+    "        data_sets = ds_importer_module.read_data_sets(ds_dataset_path, \\\n",
```
Comment (tilmankamp) [inline DeepSpeech.ipynb]: During refactoring for the last big PR, I thought about splitting the `ds_importer_module.read_data_sets` into separate calls - one per set. It could save us quite some redundancy. Now we are reading all sets for all three different set executions train/dev/test. I don't now what the exact overhead is, but at least the code would get a bit simpler.

File: util/importers/ldc93s1.py [hunk 2]
```diff
@@ -95,15 +95,17 @@ def total_batches(self):
...
+def read_data_sets(data_dir, train_batch_size, dev_batch_size, test_batch_size, numcep, numcontext, thread_count=1, limit_dev=0, limit_test=0, limit_train=0):
     # Conditionally download data
     LDC93S1_BASE = "LDC93S1"
     LDC93S1_BASE_URL = "https://catalog.ldc.upenn.edu/desc/addenda/"
     local_file = base.maybe_download(LDC93S1_BASE + ".wav", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".wav")
     _ = base.maybe_download(LDC93S1_BASE + ".txt", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".txt")
 
     # Create all DataSets, we do not really need separation
-    train = dev = test = _read_data_set(data_dir, thread_count, batch_size, numcep, numcontext)
+    train = _read_data_set(data_dir, thread_count, train_batch_size, numcep, numcontext)
```
Comment (tilmankamp) [inline util/importers/ldc93s1.py]: During refactoring for the last big PR, I thought about splitting the ds_importer_module.read_data_sets into separate calls - one per set. It could save us quite some redundancy. Now we are reading all sets for all three different set executions train/dev/test. I don't now what the exact overhead is, but at least the code would get a bit simpler.

Here we could somehow make a public version of `_read_data_set`

**Comment by kdavis-mozilla** [comment]
I think I'm going to just do the changes here, maybe later split into train, dev, and test calls.
